### PR TITLE
Polkadot v0.9.12 v2

### PR DESCRIPTION
**Pull request by jasl**
This is a WIP PR

- Can compile PhalaNode & Pherry, but not test yet
- PRuntime hasn't test compiling & functionality

### CODE REVIEW

File: crates/phala-trie-storage/src/lib.rs [hunk 1]
```diff
@@ -87,11 +87,10 @@ where
 
     /// Apply storage changes calculated from `calc_root_if_changes`.
     pub fn apply_changes(&mut self, root: H::Out, transaction: MemoryDB<H>) {
-        self.0.backend_storage_mut().consolidate(transaction);
```
Comment (h4x3rotab) [inline crates/phala-trie-storage/src/lib.rs:90]: @kvinwang

File: subxt/src/events.rs [hunk 2]
```diff
@@ -157,7 +157,6 @@ impl<T: Runtime + System> EventsDecoder<T> {
             let mut event_data = Vec::<u8>::new();
             let mut event_errors = Vec::<RuntimeError>::new();
             let result = self.decode_raw_bytes(
-                module.name(),
```
Comment (h4x3rotab) [inline subxt/src/events.rs:160]: Need to re-apply some of our changes

File: crates/phala-trie-storage/src/lib.rs (hunk 1 above)
Comment (jasl) [reply crates/phala-trie-storage/src/lib.rs:90]: forgot to port Kevin's change, will do

File: subxt/Cargo.toml [hunk 3]
```diff
@@ -50,7 +51,7 @@ sp-runtime = { path = "../substrate/primitives/runtime" }
 sp-std = { path = "../substrate/primitives/std" }
 sp-version = { path = "../substrate/primitives/version" }
 
-frame-metadata = { path = "../substrate/frame/metadata" }
+frame-metadata = { git = "https://github.com/paritytech/frame-metadata.git" }
```
Comment (h4x3rotab) [inline subxt/Cargo.toml:54]: Fix to some branch / commit?

File: crates/phala-trie-storage/src/lib.rs (hunk 1 above)
Comment (jasl) [reply crates/phala-trie-storage/src/lib.rs:90]: his port from https://github.com/Phala-Network/phala-blockchain/commit/8fb04ac61aecbeb98a7074c6e06cd8e34528d96d

File: subxt/Cargo.toml (hunk 3 above)
Comment (jasl) [reply subxt/Cargo.toml]: use crates.io instead of git now

File: subxt/src/events.rs (hunk 2 above)
Comment (jasl) [reply subxt/src/events.rs]: I've done this
https://github.com/Phala-Network/phala-blockchain/pull/553/commits/d0a65bfa15a94d3b3af0bcce37b368ac1acbdd35

I checked https://github.com/paritytech/subxt/pull/302/files?authenticity_token=VNjJrTCHNqs6JXze5OT%2FG6DfmHPq0jDobV5bQQmmwR0wFL%2BxMZsEeuY9pX3COvPh7Ymstnl3Oq7sC%2BUy83NpOA%3D%3D&file-filters%5B%5D=.rs#diff-59e0fecb904d610f27811d416aa3f26cc0d1bfed220ac69f8e18147f9bd68fd0 doesn't change events.rs so it's safe to revert the file

**Review by kvinwang — APPROVED** [review]

File: standalone/runtime/src/lib.rs [hunk 4]
```diff
@@ -1648,10 +1648,10 @@ mod tests {
 	#[test]
 	fn call_size() {
 		assert!(
-			core::mem::size_of::<Call>() <= 200,
+			core::mem::size_of::<Call>() <= 300,
```
Comment (h4x3rotab) [inline standalone/runtime/src/lib.rs:1651]: Please change the description in the next line as well

**Review by h4x3rotab — APPROVED** [review]

**Pull request merged**